pkg/pillar/zedcloud/lookupproxy.go

```diff
@@ -121,3 +121,56 @@ func LookupProxy(status *types.DeviceNetworkStatus, ifname string,
 	log.Infof("LookupProxy: No proxy configured for port %s", ifname)
 	return nil, nil
 }
+
+// IntfLookupProxyCfg - check if the intf has proxy configured
+func IntfLookupProxyCfg(status *types.DeviceNetworkStatus, ifname, downloadURL string) string {
+	var mayChange bool
+	// if proxy is not on the intf, then don't change anything
+	// if download URL has "http://" or "https://" then no change here regardless of proxy
+	// if there is proxy on this intf, treeat empty url scheme as for https,
+	// treat passed in download url schem docker://" and maybe sftp://just as https
+	if strings.Contains(downloadURL, "http") {
+		return downloadURL
+	} else if strings.Contains(downloadURL, "docker:") {
+		mayChange = true
+	}
+
+	var scheme string
+	for _, port := range status.Ports {
+		if ifname != port.IfName {
+			continue
+		}
+		var hasHTTP, hasHTTPS bool
+		for _, proxy := range port.ProxyConfig.Proxies {
+			switch proxy.Type {
+			case types.NPT_HTTP:
+				hasHTTP = true
+			case types.NPT_HTTPS:
+				hasHTTPS = true
+			}
+		}
+		if hasHTTPS {
+			scheme = "https://"
+		} else if hasHTTP {
+			scheme = "http://"
+		}
+		break
+	}
+
+	if !mayChange { // empty url scheme in downloadURL
+		return scheme + downloadURL
+	} else if scheme == "" { // if don't have proxy on intf, return original, no change
+		return downloadURL
+	}
+
+	// intf has proxy, may need to replace scheme for https with proxy cert case
+	passURL, err := url.Parse(downloadURL)
+	if err != nil {
+		return downloadURL
+	}
+	downloadURL = passURL.Host
+	if passURL.Path != "" {
+		downloadURL += ("/" + passURL.Path)
+	}
+	return scheme + downloadURL
```

Comment:
Better to do
  passURL.Scheme = scheme
  return passURL.String()

Reply:
ok. good to know.